dev-clean/headers/llsc/external_pure.hpp

```diff
@@ -74,28 +74,28 @@ inline std::string get_string(PtrVal ptr, SS state) {
   return name;
 }
 
-inline immer::flex_vector<std::pair<SS, PtrVal>> open(SS state, immer::flex_vector<PtrVal> args) {
-  PtrVal ptr = args.at(0);
-  std::string name = get_string(ptr, state);
-  FS& fs = state.get_fs();
-  /* TODO: add flags for open_file <2021-11-03, David Deng> */
-  Fd fd = fs.open_file(name, 0);
-  state.set_fs(fs);
-  return immer::flex_vector<std::pair<SS, PtrVal>>{{state, make_IntV(fd)}};
-}
-
 inline immer::flex_vector<std::pair<SS, PtrVal>> close(SS state, immer::flex_vector<PtrVal> args) {
   Fd fd = proj_IntV(args.at(0));
-  FS& fs = state.get_fs();
+  FS fs = state.get_fs();
   int status = fs.close_file(fd);
   state.set_fs(fs);
   return immer::flex_vector<std::pair<SS, PtrVal>>{{state, make_IntV(status)}};
 }
 
 inline immer::flex_vector<std::pair<SS, PtrVal>> sym_exit(SS state, immer::flex_vector<PtrVal> args) {
+  auto v = args.at(0)->to_IntV();
+  auto status = v ? v->i : 0;
```

Comment:
what's the contract of `sym_exit`? If `sym_exit` indeed requires that its first argument should be an integer, then a non-IntV argument should lead to a violation of such contract, blaming the caller and `sym_exit` should not handle the case that `v` is empty. Otherwise, if `sym_exit` should accept non-IntV value as argument, what's the point of discarding the argument and simply set the status to 0?